src/main/java/org/dcsa/ovs/notifications/service/impl/NotificationEndpointServiceImpl.java

```diff
@@ -85,12 +93,22 @@ public NotificationEndpointRepository getRepository() {
                         log.debug("Rejecting message because: " + signatureResult.getResult());
                         return Mono.error(new ResponseStatusException(HttpStatus.UNAUTHORIZED));
                     }
-                    // TODO: Implement as a part of DDT-114 using signatureResult.getParsed() and return "Mono.empty()"
-                    // (or end with .then())
-                    // Remember you can change the 4th parameter of "verifyRequest" above to change the type of
-                    // value returned by getParsed (it might require changes if you want to support complex
-                    // types)
-                    return Mono.error(new ResponseStatusException(HttpStatus.NOT_IMPLEMENTED));
+
+                    // TODO: if Event is not found, verify that we have every mandatory field.
+
+                    Mono<? extends Event> result = Mono.empty();
+                    for (Event event : signatureResult.getParsed()) {
+                        if (event instanceof OperationsEvent) {
+                             result = result.then(operationsEventRepository.findById(event.getEventID()).switchIfEmpty(operationsEventRepository.save((OperationsEvent) event)));
```

Comment:
Is it a good idea to go directly to the repository? I think we need to call the save on the service class on all these calls...